argilla-server/src/argilla_server/api/schemas/v1/users.py

```diff
@@ -40,11 +59,21 @@ class User(BaseModel):
 
 
 class UserCreate(BaseModel):
-    username: str = Field(..., min_length=1)
-    password: str = Field(min_length=USER_PASSWORD_MIN_LENGTH, max_length=USER_PASSWORD_MAX_LENGTH)
-    first_name: constr(min_length=1, strip_whitespace=True)
-    last_name: Optional[constr(min_length=1, strip_whitespace=True)] = None
+    first_name: UserFirstName
+    last_name: Optional[UserLastName] = None
+    username: UserUsername
+    role: Optional[UserRole] = None
+    password: UserPassword
+
+
+class UserUpdate(UpdateSchema):
+    __non_nullable_fields__ = {"first_name", "username", "role", "password"}
```

Comment:
We usually set `__non_nullable_fields__` after defining the attributes but I think it's ok this way too.